src/cryptsetup/cryptsetup.c

```diff
@@ -2287,6 +2343,15 @@ static int run(int argc, char *argv[]) {
                         if (r < 0)
                                 return log_error_errno(r, "Failed to load LUKS superblock on device %s: %m", crypt_get_device_name(cd));
 
+/* since cryptsetup 2.7.0 (Jan 2024) */
+#if HAVE_CRYPT_SET_KEYRING_TO_LINK
+                        if (arg_link_key_description && streq(crypt_get_type(cd), CRYPT_LUKS2)) {
```

Comment:
i'd drop the LUKS2 check from this.

I mean, if we try this on non-LUKS2 this will just result in some error code returned to us, right? which is entirely fine.

i.e. i'd not encode the feature set of libcryptsetup in our codebase needlessly. If libcryptsetup starts supporting this for other volume types this should just start to work without any changes in our codebase i think